### ISSUE TITLE
08_dataflow [optional] Setup Dataflow Development environment

### ISSUE THREAD

**Issue by pagestarr66** [GoogleCloudPlatform/data-science-on-gcp#61]
CreateTrainingDataset won't execute on my Windows machine.  I believe this is related to credentials.  I set GOOGLE_APPLICATION_CREDENTIALS to my json file but am still getting same error.

Error is at following line of code (240 in org.apache.beam.runners.dataflow.DataflowRunner:
    try {
      gcpTempLocation = dataflowOptions.getGcpTempLocation();
    } 

Called from line 96 in CreateTrainingDataset.java:
    Pipeline p = Pipeline.create(options);
-------------------------------------------------------------------------------------------------------
Pipeline options are below:
  appName: CreateTrainingDataset
  bucket: data_science_on_gcp
  credentialFactoryClass: class org.apache.beam.sdk.extensions.gcp.auth.GcpCredentialFactory
  gcpCredential: null
  gcsEndpoint: null
  optionsId: 0
  pathValidator: org.apache.beam.sdk.extensions.gcp.storage.GcsPathValidator@5cbe877d
  pathValidatorClass: class org.apache.beam.sdk.extensions.gcp.storage.GcsPathValidator
  project: neural-virtue-236312
  runner: class org.apache.beam.runners.dataflow.DataflowRunner
  stableUniqueNames: WARNING
  tempLocation: gs://data_science_on_gcp/flights/staging

---------------------------------------------------------------------------------------------------
Maven output below:

Current Settings:
  appName: CreateTrainingDataset
  bucket: data_science_on_gcp
  optionsId: 0
  project: neural-virtue-236312
  runner: class org.apache.beam.runners.dataflow.DataflowRunner
  stableUniqueNames: WARNING
  tempLocation: gs://data_science_on_gcp/flights/staging
Exception in thread "main" java.lang.RuntimeException: Failed to construct instance from factory method DataflowRunner#fromOptions(interface org.apache.beam.sdk.options.PipelineOptions)
	at org.apache.beam.sdk.util.InstanceBuilder.buildFromMethod(InstanceBuilder.java:233)
	at org.apache.beam.sdk.util.InstanceBuilder.build(InstanceBuilder.java:162)
	at org.apache.beam.sdk.PipelineRunner.fromOptions(PipelineRunner.java:55)
	at org.apache.beam.sdk.Pipeline.create(Pipeline.java:150)
	at com.google.cloud.training.flights.CreateTrainingDataset.main(CreateTrainingDataset.java:96)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.apache.beam.sdk.util.InstanceBuilder.buildFromMethod(InstanceBuilder.java:222)
	... 4 more
Caused by: java.lang.NoSuchMethodError: com.google.api.client.googleapis.services.json.AbstractGoogleJsonClient$Builder.setBatchPath(Ljava/lang/String;)Lcom/google/api/client/googleapis/services/AbstractGoogleClient$Builder;
	at com.google.api.services.storage.Storage$Builder.setBatchPath(Storage.java:9307)
	at com.google.api.services.storage.Storage$Builder.<init>(Storage.java:9286)
	at org.apache.beam.sdk.util.Transport.newStorageClient(Transport.java:95)
	at org.apache.beam.sdk.util.GcsUtil$GcsUtilFactory.create(GcsUtil.java:96)
	at org.apache.beam.sdk.util.GcsUtil$GcsUtilFactory.create(GcsUtil.java:84)
	at org.apache.beam.sdk.options.ProxyInvocationHandler.returnDefaultHelper(ProxyInvocationHandler.java:592)
	at org.apache.beam.sdk.options.ProxyInvocationHandler.getDefault(ProxyInvocationHandler.java:533)
	at org.apache.beam.sdk.options.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:155)
	at com.sun.proxy.$Proxy24.getGcsUtil(Unknown Source)
	at org.apache.beam.sdk.extensions.gcp.storage.GcsPathValidator.verifyPathIsAccessible(GcsPathValidator.java:88)
	at org.apache.beam.sdk.extensions.gcp.storage.GcsPathValidator.validateOutputFilePrefixSupported(GcsPathValidator.java:61)
	at org.apache.beam.sdk.extensions.gcp.options.GcpOptions$GcpTempLocationFactory.create(GcpOptions.java:245)
	at org.apache.beam.sdk.extensions.gcp.options.GcpOptions$GcpTempLocationFactory.create(GcpOptions.java:228)
	at org.apache.beam.sdk.options.ProxyInvocationHandler.returnDefaultHelper(ProxyInvocationHandler.java:592)
	at org.apache.beam.sdk.options.ProxyInvocationHandler.getDefault(ProxyInvocationHandler.java:533)
	at org.apache.beam.sdk.options.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:155)
	at com.sun.proxy.$Proxy15.getGcpTempLocation(Unknown Source)
	at org.apache.beam.runners.dataflow.DataflowRunner.fromOptions(DataflowRunner.java:240)
	... 9 more

**Comment by lakshmanok** [comment]
Change the bucket to be something that you own.

**Comment by pagestarr66** [comment]
Hi thanks for responding.  I do own this bucket, further dets are below.  I set it up to go through the chapters in this book (at the time I didn't realise this name would be unique across GCP).  

Created	1 April 2019 at 13:12:07 UTC+1
Updated	1 April 2019 at 13:12:07 UTC+1
Location	 europe-west2 (London)
Default storage class	Regional
Access control model	Permissions set at object level (ACL) and bucket level (IAM)
Requester pays	        Off
Encryption type	        Google-managed key
Link URL	                         https://console.cloud.google.com/storage/browser/data_science_on_gcp
Link for gsutil	                 gs://data_science_on_gcp
